src/test/rustdoc/cross-crate-hidden.rs

```diff
@@ -0,0 +1,35 @@
+// Issue #86448: test for cross-crate `doc(hidden)`
+#![crate_name = "foo"]
+
+// aux-build:cross-crate-hidden.rs
+extern crate cross_crate_hidden;
+
+pub use ::cross_crate_hidden::{HiddenType, HiddenTrait}; // OK, not re-exported
```

Comment:
Can you also test this for the same-crate case?
```suggestion
// revisions: extern local
// [extern] aux-build:cross-crate-hidden.rs
#[cfg(extern)]
extern crate cross_crate_hidden;
#[cfg(extern)]
pub use ::cross_crate_hidden::{HiddenType, HiddenTrait}; // OK, not re-exported

#[cfg(local)]
#[doc(hidden)]
pub enum HiddenType {}

#[cfg(local)]
#[doc(hidden)]
pub trait HiddenTrait {}
```
(not sure if revisions work for rustdoc tests - if not, it's fine to make these two separate tests.)

Please also change the file name from `cross-crate-hidden` to `cross-crate-hidden-impl-parameter`, which I think is a more accurate description.

Reply:
revisions do not work - I have added a separate test